optuna/storages/_in_memory.py

```diff
@@ -335,6 +346,40 @@ def set_trial_values(self, trial_id: int, values: Sequence[float]) -> None:
             trial.values = values
             self._set_trial(trial_id, trial)
 
+    def set_trial_state_values(
+        self, trial_id: int, state: TrialState, values: Optional[Sequence[float]] = None
+    ) -> bool:
+
+        with self._lock:
+            trial = self._get_trial(trial_id)
+            self.check_trial_is_updatable(trial_id, trial.state)
+
+            trial = copy.copy(trial)
+            self.check_trial_is_updatable(trial_id, trial.state)
+
+            if values is not None:
+                trial.values = values
```

Comment:
If values are not None, we always set values.
After this line, we can write `self._set_trial(trial_id, trial)`?

Reply:
It makes code more simple, but when updating state we call `self._set_trial(trial_id, trial)` twice. So I selected more effective way. Which do you like ?

Reply:
I understand. Leave this line as it is.